contracts/protocol/pool/Pool.sol

```diff
@@ -146,7 +146,7 @@ contract Pool is VersionedInitializable, IPool, PoolStorage {
     uint256 amount,
     address onBehalfOf,
     uint16 referralCode
-  ) external virtual override {
+  ) public virtual override {
```

Comment:
Why public instead of internal? 
I assume the idea is that the L2Pool interface will only be called by EOAs and this interface will be used by contracts on L2?

One concern might be that the `public` keyword increases gas costs for those other contracts.

Reply:
Your assumption is correct. The compressed versions of the functions that are available in the L2Pool is only planned to be called by EOAs. 
Contracts are expected to just use the Pool as on any L1. The reason behind this is that the cost of L2 transactions are primarily denominated by the calldata they need to put on L1. In the optimistic rollups, this would be the input from an EOA, which is compressed heavily in L2Pool. For the case where an L2contract calls another L2contract this influence the L2 gas-cost, but not the data that needs to be published on L1, which means that we do not gain much from also doing the compression in the "middle". In essence, this means that contracts and integrators on L2 can implement similarly to L1 deployments, but frontends or direct calls can save some gas by using the compressed version.

Reply:
Yes, though I was referring to the fact that the `public` kw can sometimes increase costs vs. `external`. So the alternative would be to make these `internal`, and add a new set of `external` functions with this interface which wrap those. 

However I realize now that `public`/`external` should only make a difference with reference types  which none of these functions use.

Reply:
This can be resolved.